File: backend/src/main/java/com/bookbook/global/security/refreshToken/RefreshTokenRepository.java
package com.bookbook.global.security.refreshToken;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

public interface RefreshTokenRepository extends JpaRepository<RefreshToken, Long> {
    Optional<RefreshToken> findByToken(String token);
    Optional<RefreshToken> findByUserId(Long userId);

    @Transactional
    void deleteByUserId(Long userId);
}

File: backend/src/main/java/com/bookbook/global/security/refreshToken/RefreshTokenRepository.kt
package com.bookbook.global.security.refreshToken

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.transaction.annotation.Transactional

interface RefreshTokenRepository : JpaRepository<RefreshToken, Long> {
    fun findByToken(token: String): RefreshToken?
    fun findByUserId(userId: Long): RefreshToken?

    @Transactional
    fun deleteByUserId(userId: Long)
}